client/analytics/components/report-filters/index.js

```diff
@@ -135,12 +139,18 @@ class ReportFilters extends Component {
 
 ReportFilters.contextType = CurrencyContext;
 
-export default withSelect( ( select ) => {
-	const { woocommerce_default_date_range: defaultDateRange } = select(
-		SETTINGS_STORE_NAME
-	).getSetting( 'wc_admin', 'wcAdminSettings' );
-	return { defaultDateRange };
-} )( ReportFilters );
+export default compose(
+	withSelect( ( select ) => {
+		const { woocommerce_default_date_range: defaultDateRange } = select(
+			SETTINGS_STORE_NAME
+		).getSetting( 'wc_admin', 'wcAdminSettings' );
+		return { defaultDateRange };
+	} ),
+	withDispatch( ( dispatch ) => {
+		const { addCesSurveyTrackForAnalytics } = dispatch( CES_STORE_KEY );
```

Comment:
I think backend and frontend APIs have a slightly different purpose in this case. The backend saves the queue in the database to be used at a later time, but the tracks added from the frontend never get saved. From the API user's (including us and a 3PD dev) point of view, the user wants to add a survey immediately instead of queueing it to be displayed later. 

However, I agree that this can be confusing for those who use both backend and frontend APIs.

cc @becdetat for additional thoughts

Reply:
That makes sense -- thanks for the clarification. But, this still isn't adding the Tracks log entry, it is just showing the CES survey, right? I think it is best to be as clear as possible in our naming in this regard so that others looking at the code don't misinterpret what is going on.

Reply:
Also, from the API user's standpoint, they are abstracted from where the survey is getting saved, right? If so, we shouldn't use the "Track" terminology.

Reply:
@mattsherman I think it tracks as the function basically re-uses the [CustomerEffortScoreTracks](https://github.com/woocommerce/woocommerce-admin/blob/add/5349-trigger-ces-on-filtering-analytics/client/customer-effort-score-tracks/customer-effort-score-tracks.js). However, I'm not sure if we want to track 3PD surveys to our system though. I think we need further discussion on that.

Reply:
There's some more background about the goals for abstraction and external use in the [umbrella issue](https://github.com/woocommerce/woocommerce-admin/issues/5259). 

I  think we should align closely with the backend naming - `addCesSurveyForAnalytics` or `enqueueCesSurveyForAnalytics` LGTM.

Reply:
@adrianduffell Thank you for the suggestion. Updated in https://github.com/woocommerce/woocommerce-admin/pull/5649/commits/c8ba3dd0c965a554610f4854557ef60ab0a3bd6f